core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -68,4 +82,8 @@ public static Table createMetadataTableInstance(TableOperations ops,
     String metadataTableName = BaseMetastoreCatalog.fullTableName(catalogName, metadataTableIdentifier);
     return createMetadataTableInstance(ops, baseTableName, metadataTableName, type);
   }
+
+  private static String metadataTableName(String tableName, MetadataTableType type) {
+    return tableName + (tableName.contains("/") ? "#" : ".") + type;
```

Comment:
Question: I have not seen any implementations that do this, but I believe that this character can be set. Technically, I think it would be possible for the whole name to be set by somebody presently, though I doubt anybody is doing that.

Should we consider either enforcing these characters or allowing for the possibility of extra tables and metadata tables to table types (e.g. maybe Glue table type for some spelling reason, etc)?